std/container/rbtree.d

```diff
@@ -1094,6 +1094,108 @@ if (is(typeof(binaryFun!less(T.init, T.init))))
         assert(t1 != o);  // pathological case, must not crash
     }
 
+    /**
+     * Generates a hash for the tree
+     */
+    override size_t toHash() nothrow @safe
+    {
+        import std.traits : hasMember;
+
+        auto thisRange = this[];
+
+        static if (hasMember!(Elem, "toHash"))
```

Comment:
Also: this should be `static if (hasMember!(Elem, "toHash") || is(Elem == interface ))` so this will work with interfaces. Demonstration:

https://run.dlang.io/is/tt68Vd
```d
interface I {}
void main()
{
    static assert(__traits(hasMember, I, "toHash")); // Fails!
}
```